test/Base_Tests/src/Data/Vector_Spec.enso

```diff
@@ -423,6 +423,12 @@ type_spec suite_builder name alter = suite_builder.group name group_builder->
         vec_mut = Vector.new_builder
         vec.each vec_mut.append
         vec_mut.to_vector . should_equal vec
+    
+    group_builder.specify "should allow errors to bubble up out of a .each" <|
+        do_write x =
+            Error.throw "Failed to write "+x.to_text
+        vec = alter [1, 2, 3, 4]
+        vec.each do_write . should_fail_with ("Failed to write 1")
```

Comment:
This is cool but it reminds us of another problem.

Because `v.each` is an inherently side-effect-oriented operation, its result is often discarded. In such case, even if `v.each` propagates the dataflow error, that dataflow error is then very likely to be discarded and ignored anyway.

Given that we do know that `v.each` result is most often ignored (it is partly by design...), we could consider promoting errors in `each` to Panics, so that they are not ignored so easily. But that ofc has the downside that it will make more panics surface. However, I'm not really sure if that is a problem...

Alternatively, I guess an ideal solution would be to finally get to https://github.com/enso-org/enso/issues/5430 to automatically promote such discarded dataflow errors to panics, or to propagate them to the return value of the function/block in which it was discarded.

While fortunately for our users not something likely to happen in the GUI, in text mode the current model of propagation of dataflow errors is going to likely lead to subtle issues when such dataflow errors are discarded.

Reply:
Fixed

Reply:
I'm not sure I know enough of the philosophy of the language around errors and panics. But it feels weird converting errors to panics.

The other thing I wondered as I made this change is should each return the original collection instead of nothing? Also not sure if that is a good idea or not...

Reply:
Errors now throw as panics